pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+                        NewClass.classname, namespace))
+
+        new_class = NewClass.copy()
+
+        qualifier_store = self.get_qualifier_store(namespace)
+        self._resolve_class(new_class, namespace, qualifier_store,
+                            verbose=False)
+
+        # Add new class to CIM repository
+        class_store.create(new_class.classname, new_class)
+
+    ####################################################################
+    #
+    #   ModifyClass server request
+    #
+    ####################################################################
+
+    def ModifyClass(self, namespace, ModifiedClass):
+        """
+        This method is not implemented and returns CIM_ERR_NOT_SUPPORTED.
+
+        Implements a  WBEM server responder method for
+        :meth:`~pywbem.WBEMConnection.MmodifyClass`
+
+        Modifies a class in the CIM repository.  Nothing is returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace containing the class that will be modified
+            (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ModifiedClass (:class:`~pywbem.CIMClass`):
+            A representation of the modified class.
+
+            The properties, methods and qualifiers defined in this object
+            specify what is to be modified.
+
+        Raises:
+
+            CIMError: CIM_ERR_NOT_SUPPORTED
+        """
+
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            "Currently ModifyClass not supported in MainProvider")
+
+    ####################################################################
+    #
+    #   DeleteClass server request
+    #
+    ####################################################################
+
+    def DeleteClass(self, namespace, ClassName):
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.DeleteClass`
+
+        Delete a class in the class CIM repository if it exists.
+
+        This method also deletes any subclasses and CIM instances of
+        the target class or its subclasses.
+
+        Nothing is returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace containing the class defined by ClassName that
+            will be deleted (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ClassName (:term:`string`):
+            Name of the class to be deleted (case independent).
+
+        Raises:
+
+            CIMError: CIM_ERR_NOT_FOUND if ClassName defines class not in
+                CIM repository
+        """
+
+        # Validate namespace and get the class datastore for this namespace
+        class_store = self.get_class_store(namespace)
+        instance_store = self.get_instance_store(namespace)
```

Comment:
I'd like to again bring up the idea to not validate the namespace in the get_..._store() methods, but to call it explicitly where needed. These two get_..._store() calls show that having the validation being a side-effect is not good. In this case, the validation takes place twice, and nevertheless it is a silent side effect and not explicit enough.

Reply:
Removed the validation call from each of the get_..._store methods and inserted a validate_namespace(namespace at the beginning of each of the original request methods. 

Note that none of the open opertions use validate_namespace() because they call the corresponding original operation and the namespace test runs there.

Reply:
great!